.github/workflows/integration-test.yml

```diff
@@ -0,0 +1,169 @@
+name: Integration Tests
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    types: [opened, synchronize, reopened]
+
+jobs:
+  test:
+    name: Test (${{ matrix.integration.name }})
+    runs-on: ${{ matrix.integration.os }}-latest
+    env:
+      MISE_PROFILE: cicd
+    strategy:
+      fail-fast: false
+      matrix:
+        integration:
+          - name: Fixtures
+            os: ubuntu
+            target: ./test
+          - name: SSH
+            os: ubuntu
+            target: ./...
+            setup_scripts:
+              - .github/scripts/setup/ssh.sh
+            tags: ssh
+            run: '^TestSSH'
+            secrets: [GHA_DEPLOY_KEY]
+          - name: SOPS
+            os: ubuntu
+            target: ./...
+            setup_scripts:
+              - .github/scripts/setup/sops.sh
+            tags: sops
+            run: '^TestSOPS'
+          - name: Tflint
+            os: ubuntu
+            target: ./...
+            setup_scripts:
+              - .github/scripts/setup/tflint.sh
+            tags: tflint
+            run: '^TestTflint'
+            secrets: [AWS_ACCESS_KEY_ID, AWS_SECRET_ACCESS_KEY]
+            # requires AWS credentials to run
+            skip: true
+          - name: GCP
+            os: ubuntu
+            target: ./...
+            setup_scripts:
+              - .github/scripts/setup/gcp.sh
+            tags: gcp
+            run: '^TestGcp'
+            secrets: [GCLOUD_SERVICE_KEY, GOOGLE_CLOUD_PROJECT, GOOGLE_COMPUTE_ZONE, GOOGLE_IDENTITY_EMAIL, GOOGLE_PROJECT_ID]
+            skip: true
+          - name: AWS
+            os: ubuntu
+            target: ./...
+            setup_scripts:
+              - .github/scripts/setup/aws.sh
+            tags: aws
+            run: '^TestAws'
+            secrets: [AWS_ACCESS_KEY_ID, AWS_SECRET_ACCESS_KEY]
+            # requires AWS credentials to run
+            skip: true
+          - name: AWSGCP
+            os: ubuntu
+            target: ./...
+            setup_scripts:
+              - .github/scripts/setup/aws.sh
+              - .github/scripts/setup/gcp.sh
+            tags: awsgcp
+            run: '^TestAwsGcp'
+            secrets: [AWS_ACCESS_KEY_ID, AWS_SECRET_ACCESS_KEY, GCLOUD_SERVICE_KEY, GOOGLE_CLOUD_PROJECT, GOOGLE_COMPUTE_ZONE, GOOGLE_IDENTITY_EMAIL, GOOGLE_PROJECT_ID]
+            # requires AWS credentials to run
+            skip: true
+          - name: Engine
+            os: ubuntu
+            target: ./...
+            setup_scripts:
+              - .github/scripts/setup/engine.sh
+            tags: engine
+            run: '^TestEngine'
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Generate Secrets Environment
+        id: generate-secrets
+        run: |
+          ENV_FILE="${GITHUB_WORKSPACE}/.env.secrets"
+          echo "export SCRIPT_PATHS='${{ matrix.integration.setup_scripts }}'" > $ENV_FILE
```

Comment:
We should never directly interpolate these in the workflow. Please forward this through an `env` to avoid attacks involving shell injection.